router/routing/app.go

```diff
@@ -70,3 +80,165 @@ func HandleCmdConfigGetExtendedReport(serviceName string, storage storage.Storag
 			return msg, nil
 		}))
 }
+
+// RouteCmdAppGetManifest returns a routing responsible for handling the command.
+func RouteCmdAppGetManifest(
+	serviceName string,
+	appLifecycle *lifecycle.Lifecycle,
+	configStorage storage.Storage,
+	manifestManager manifest.Manager,
+) *router.Routing {
+	return router.NewRouting(
+		HandleCmdAppGetManifest(serviceName, appLifecycle, configStorage, manifestManager),
+		router.ForService(serviceName),
+		router.ForType(CmdAppGetManifest),
+	)
+}
+
+// HandleCmdAppGetManifest returns a handler responsible for handling the command.
+func HandleCmdAppGetManifest(
+	serviceName string,
+	appLifecycle *lifecycle.Lifecycle,
+	configStorage storage.Storage,
+	manifestManager manifest.Manager,
+) router.MessageHandler {
+	return router.NewMessageHandler(
+		router.MessageProcessorFn(func(message *fimpgo.Message) (*fimpgo.FimpMessage, error) {
+			mode, err := message.Payload.GetStringValue()
+			if err != nil {
+				return nil, fmt.Errorf("provided value has an incorrect format: %w", err)
+			}
+
+			m, err := manifestManager.Get()
+			if err != nil {
+				return nil, fmt.Errorf("failed to retrieve the manifest: %w", err)
+			}
+
+			if mode == "manifest_state" {
+				m.AppState = *appLifecycle.GetAllStates()
+				m.ConfigState = configStorage.Model()
+			}
+
+			reply := fimpgo.NewMessage(EvtAppManifestReport, serviceName, fimpgo.VTypeObject, m, nil, nil, message.Payload)
+
+			return reply, nil
+		}),
+	)
+}
+
+// RouteCmdConfigExtendedSet returns a routing responsible for handling the command.
+// Provided locker is optional.
+func RouteCmdConfigExtendedSet(
+	serviceName string,
+	appLifecycle *lifecycle.Lifecycle,
+	configFactory func() interface{},
+	manifestManager manifest.Manager,
+	locker router.MessageHandlerLocker,
+) *router.Routing {
+	return router.NewRouting(
+		HandleCmdConfigExtendedSet(serviceName, appLifecycle, configFactory, manifestManager, locker),
+		router.ForService(serviceName),
+		router.ForType(CmdConfigExtendedSet),
+	)
+}
+
+// HandleCmdConfigExtendedSet returns a handler responsible for handling the command.
+// Provided locker is optional.
+func HandleCmdConfigExtendedSet(
+	serviceName string,
+	appLifecycle *lifecycle.Lifecycle,
+	configFactory func() interface{},
```

Comment:
Why `configFactory` instead just `config`?

Reply:
Because you don't want to directly unmarshal provided value in an incoming message into config, which is used by the entire application without concurrency safeguards and your validation of input. What the handler does later is as follows: it passes a fresh instance of a config with data hydrated from the incoming message to the manager, for the manager to decide what to do with it. You may decide that the data that came in is outside of accepted ranges.

Think about it in a context of a form. You want to map a form to an object and then decide which fields from this object to use and which don't.

Reply:
Ok, understand the point.